torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -449,6 +484,99 @@ void ProcessGroupAgent::listenLoop() {
   }
 }
 
+void ProcessGroupAgent::pollTimedOutRPCs() {
+  while (!shutdown_.load()) {
+    std::chrono::milliseconds sleepTime;
+    std::unique_lock<std::mutex> lock{futureMutex_};
+    // Estimate amount of time the first future will time out in, and sleep
+    // for that long.
+    // if there are no futures or the first future's RPC timeout is set to 0
+    // (meaning no timeout), then sleep for a set "infinity" time.
+    if (futureTimeouts_.empty() ||
+        futureTimeouts_.begin()->first == INFINITE_TIMEOUT) {
+      sleepTime = INFINITE_TIMEOUT;
+    } else {
+      const auto minFutureExpirationTime = futureTimeouts_.begin()->first;
+      const auto remainingTime = getRPCRemainingTime(minFutureExpirationTime);
+      sleepTime = std::max(remainingTime, std::chrono::milliseconds(0));
+    }
+
+    if (sleepTime == INFINITE_TIMEOUT) {
+      futureTimeoutCV_.wait_until(
+          lock, std::chrono::time_point<std::chrono::system_clock>::max());
+    } else {
+      futureTimeoutCV_.wait_for(lock, sleepTime);
```

Comment:
I encountered some overflow errors when trying to pass in `INFINITE_TIMEOUT` into `wait_for`. Not sure if there's a better way to get around this.

Reply:
In this case, I think it's idiomatic to just use wait():

  if (sleepTime == INFINITE_TIMEOUT) {
       futureTimeoutCV_.wait();
  } else {
     futureTimeoutCV_.wait_for(lock, sleepTime);
  }